eureka-server/src/main/java/com/netflix/eureka2/server/channel/InterestChannelImpl.java

```diff
@@ -53,9 +60,24 @@ public void call(Object message) {
                  * invocation), we do not need to worry about thread safety here and hence we can safely do a
                  * isRegistered -> register kind of actions without worrying about race conditions.
                  */
-                if (message instanceof InterestRegistration) {
+                if (message instanceof SnapshotRegistration) {
+                    switch (state.get()) {
+                        case Idle:
+                            sendSnapshot(((SnapshotRegistration) message).getInterests());
+                            state.set(STATES.Closed);
+                            metrics.stateTransition(STATES.Idle, STATES.Open);
```

Comment:
What is the state diagram for the snapshot request? The previous line sets the state to Closed but the metric transition only moves from Idle to Open.